pkg/rules/engine/engine.go

```diff
@@ -307,3 +307,57 @@ func (engine *Engine) GetSelectedEvents() []detect.SignatureEventSelector {
 	}
 	return res
 }
+
+// GetFilters returns a mediated array of all signature's filters
```

Comment:
might be helpful to give a short explanation on what mediated array is.

Reply:
Well mediation is a very general concept. But mediated has a meaning, how the mediation itself is done could be explained later in the function. But for mediation examples, one can look at the tests.